lib/compilers/dotnetcore.js

```diff
@@ -0,0 +1,120 @@
+</Project>
+`;
+
+class DotNetCoreCompiler extends BaseCompiler {
+
+    constructor(info, env) {
+        super(info, env);
+        this.asm = new AsmParser(this.compilerProps, {
+            labelDef: /^([.A-Za-z_$@][A-Za-z0-9$_@.]*):/i,
+            commentOnly: /^\s*(((#|@|;|\/\/)((?!Assembly listing for method).)*)|(\/\*.*\*\/))$/,
+            removeBlankLines: true,
+        });
+    }
+
+    optionsForFilter(filters, outputFilename) {
+        filters.preProcessLines = (asmLines) => this.preProcessLines(asmLines);
+        return [outputFilename];
+    }
+
+    preProcessLines(asmLines) {
+        for (const [index, line] of asmLines.entries()) {
+            // remove hex dump of each opcode that precedes the assembly
+            const hexDump = /^( +)([A-Z]|[0-9])*( *)/;
+            asmLines[index] = line.replace(hexDump, (_, spaces) => spaces);
+
+            if (line.startsWith('Microsoft (R) CoreCLR Native Image Generator')) {
+                asmLines.length = index;
+                break;
+            }
+        }
+        return asmLines;
+    }
+
+    runCompiler(compiler, options, inputFilename, execOptions) {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        execOptions.env = Object.assign({}, execOptions.env);
+        for (const [env, to] of this.compiler.envVars) {
+            execOptions.env[env] = to;
+        }
+
+        execOptions.customCwd = path.dirname(inputFilename);
+
+        const projectFilePath = path.join(execOptions.customCwd, "Project.csproj");
+        const outputPath = this.filename(options[0]);
+
+        options = ['publish', '-c', BuildConfiguration, '--self-contained'];
+
+        const coreClrPath = execOptions.env['CoreCLRDebugPath'];
+        const publishPath = path.join(
+            execOptions.customCwd, 'bin', BuildConfiguration,
+            TargetFramework, RuntimeIdentifier, 'publish');
+        const crossgenPath = path.join(publishPath, 'crossgen');
+        const dllPath = path.join(publishPath, 'Project.dll');
+
+        return this.writeFile(projectFilePath, ProjectFileContent).
+            then(() => super.runCompiler(compiler, options, inputFilename, execOptions).
+            // copy the debug CoreCLR executables (they need to be in the same directory as the DLL for some reason)
+            then(compilerRet => fs.copy(coreClrPath, publishPath). 
+            then(() => this.runCrossgen(execOptions, crossgenPath, publishPath, dllPath, outputPath)).
+            then(() => compilerRet)));
+    }
+
+    runCrossgen(execOptions, crossgenPath, publishPath, dllPath, outputPath) {
+        // run crossgen (managed to native)
+        // set these environment variables to activate the disassembly output
+        execOptions.env['COMPlus_NgenDisasm'] = '*';
+        execOptions.env['COMPlus_NgenDiffableDasm'] = '1';
+        return exec.execute(crossgenPath, ['/Platform_Assemblies_Paths', publishPath, dllPath], execOptions).
```

Comment:
There are a few more options I'd find useful, e.g. `COMPlus_JitGCDump`, `COMPlus_JitUnwindDump`, and `COMPlus_JitEHDump`. Of course, this could be addressed later.

One question is whether it'd be possible to support setting environment variables. There are more JIT knobs that domain experts may want to play with.